Tests/RxSwiftTests/SingleTest.swift

```diff
@@ -585,6 +585,30 @@ extension SingleTest {
             .completed(200)
             ])
     }
+
+    func test_asCompletable() {
+        let scheduler = TestScheduler(initialClock: 0)
+
+        let res = scheduler.start {
+            (Single<Int>.just(5).asCompletable() as Completable).asObservable()
```

Comment:
I think the multiple casting and hinting here make the test non-deterministic.
e.g. making the Completable an Observable. Not sure if there's a nicer way around this but that's my personal opinion at least.

Reply:
The problem is that the closure from `scheduler.start` returns an `Observable`. 
I could call `subscribe` on my own, store the events on an array and check the received events, but this sounds so bad 😝 
All tests on traits end with this `asObservable` call. I really don't how we can make this better.

Reply:
Gotta. That's fair I think. If Kruno dislikes this he'll review as well.